File: classes01/python/trackbar.py
import cv2

threshold_value=200
threshold_type=3
max_value=255
max_type=4
max_BINARY_value=255

windowName = "Threshold Demo"
trackbarType = "Type: \n 0: Binary \n 1: Binary Inverted \n 2: Truncate \n 3: To Zero \n 4: To Zero Inverted"
trackbarValue = "Value"

imGray = cv2.imread("../../data/images/threshold.png", cv2.IMREAD_GRAYSCALE)
cv2.namedWindow(windowName, cv2.WINDOW_AUTOSIZE)

def thresholdTypeDemo(*args):
    global threshold_type
    threshold_type = args[0]
    thresh, result = cv2.threshold(imGray, threshold_value, max_BINARY_value, threshold_type)
    cv2.imshow(windowName, result)


def thresholdValueDemo(*args):
    global threshold_value
    threshold_value = args[0]
    thresh, result = cv2.threshold(imGray, threshold_value, max_BINARY_value, threshold_type)
    cv2.imshow(windowName, result)

cv2.createTrackbar(trackbarType, windowName, threshold_type, max_type, thresholdTypeDemo)

cv2.createTrackbar(trackbarValue, windowName, threshold_value, max_value, thresholdValueDemo)

thresholdValueDemo(0)

# Wait until user finishes program
while True:
    c = cv2.waitKey(20)
    if c == 27:
        break

cv2.destroyAllWindows()

File: classes01/cpp/trackbar.cpp
#include <opencv2/opencv.hpp>
#include <string>

using namespace std;
using namespace cv;

int thresholdValue=150;
int thresholdType=3;
int const maxValue=255;
int const maxType=4;
int const max_BINARY_value=255;

Mat im, imGray, dst;

string windowName="Threshold Demo";
string trackbarType="Type: \n 0: Binary \n 1: Binary Inverted \n 2: Truncate \n 3: To Zero \n 4: To Zero Inverted";
string trackbarValue="Value";

/**
 * @brief 
 * 
 */
void thresholdDemo(int,void*);

int main()
{
    im=imread("../../data/images/threshold.png");

    cvtColor(im,imGray,COLOR_RGB2GRAY);

    namedWindow(windowName,CV_WINDOW_AUTOSIZE);

    createTrackbar(trackbarType, windowName, &thresholdType, maxType, thresholdDemo);

    thresholdDemo(0,0);

    while(true)
    {
        int c;
        c = waitKey(20);
        if(c==27)
         break;
        
    }

    return EXIT_SUCCESS;
}

void thresholdDemo(int,void*)
{
    threshold(imGray, dst, thresholdValue, max_BINARY_value, thresholdType);
    imshow(windowName, dst);
}